cmd/kubeapps-apis/plugins/resources/v1alpha1/namespaces.go

```diff
@@ -112,7 +114,14 @@ func (s *Server) CanI(ctx context.Context, r *v1alpha1.CanIRequest) (*v1alpha1.C
 	}
 	log.InfoS("+resources CanI", "cluster", cluster, "namespace", namespace, "group", r.GetGroup(), "resource", r.GetResource(), "verb", r.GetVerb())
 
-	typedClient, _, err := s.clientGetter(ctx, cluster)
+	var typedClient kubernetes.Interface
+	var err error
+	if s.kubeappsCluster != cluster && strings.ToLower(r.GetVerb()) == "list" && strings.ToLower(r.GetResource()) == "namespaces" {
+		// Listing namespaces in additional clusters might involve using the provided service account token
+		typedClient, _, err = s.clusterServiceAccountClientGetter(ctx, cluster)
```

Comment:
If actual listing of namespaces (` GetAccessibleNamespaces`) makes use of the `clusterServiceAccountClientGetter` for remote clusters, it makes sense that `CanI` behaves accordingly.